jetbrains-core/tst/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialogTest.kt

```diff
@@ -64,12 +96,89 @@ class RunCommandDialogTest {
         )
         val samplePath = tempFolder.newFile("sample-file").toPath()
         runInEdtAndWait {
-            val execCommand = RunCommandDialog(projectRule.project, container).buildExecCommandConfiguration(command, samplePath)
+            val execCommand = RunCommandDialog(projectRule.project, container, connectionSettings).buildExecCommandConfiguration(command, samplePath)
             assertThat(execCommand.name).isEqualTo("sample-container")
             assertThat(execCommand.parameters).isEqualTo(verifyCommand)
             assertThat(execCommand.isUseConsole).isTrue
             assertThat(execCommand.isShowConsoleOnStdOut).isTrue
             assertThat(execCommand.program).isEqualTo(samplePath.toString())
         }
     }
+
+    @Test
+    fun `Credentials are attached as environment variables when running AWS CLI`() {
+        resourceCache.addEntry(
+            projectRule.project, EcsResources.listTasks(clusterArn, serviceArn),
+            CompletableFuture.completedFuture(taskList)
+        )
+        val programPath = makeSampleCliExecutable()
+        runInEdtAndWait {
+            val environmentVariables = mutableListOf<String>()
+            val counter = CountDownLatch(5)
+            val execCommand = RunCommandDialog(projectRule.project, container, connectionSettings)
+                .buildExecCommandConfiguration(('"' + command + '"'), programPath)
+            val environment =
+                ExecutionEnvironmentBuilder
+                    .create(
+                        projectRule.project,
+                        DefaultRunExecutor.getRunExecutorInstance(),
+                        RunCommandRunProfile(execCommand, SimpleDataContext.getProjectContext(projectRule.project), connectionSettings)
+                    )
+                    .build {
+                        it.processHandler?.addProcessListener(object : ProcessAdapter() {
+
+                            override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                                super.onTextAvailable(event, outputType)
+                                environmentVariables.add(event.text.split("\n").first())
+                                counter.countDown()
```

Comment:
You can move this to the onProcessTerminated and not tie the countdown to the number of lines to make the test less rigid

Reply:
done